File: Code/Matlab/verifyNeighbors.m
function [neighbor]=verifyNeighbors(p,thres,all)
% return all indices whose value are no more than the minimal value by
% thres; if there are more than one such scale, return the largest possible
if nargin<2
    thres=0;
end
if nargin<3
    all=0; % if all is not 0, return all indices; if all is 0, return only the largest possible index.
end
n=size(p,1);
pmin=min(min(p));

nind=find((p-pmin)<=thres);
neighbor=nind(1);

if all==0
    max=0;
    for i=1:length(nind)
        l=ceil(nind(i)/n);
        k=nind(i)-(l-1)*n;
        if (k*l)>=max
            neighbor=nind(i);
            max=k*l;
        end
    end
end

    

File: Code/R/verifyNeighbors.R
#source("verifyNeighbors.R")

verifyNeighbors <-function(p,thres){
  if (missing(thres)){
    thres=0; 
  }
  n=nrow(p);
  pmin=min(p);
  
  neighbor=which((p-pmin)<=thres);
  #neighbor=matrix(0,length(nind),2);
  #nind=nind[length(nind)];
  
  #   ind=((p-pmin)<thres);
  #   nind=which(ind==TRUE);
  #   nind=nind[length(nind)];
  #   
  #neighbor[,2]=ceiling(nind/n);
  #neighbor[,1]=nind-(neighbor[,2]-1)*(n);
  return(neighbor);
}